src/iqm/qiskit_iqm/fake_backends/fake_deneb.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2022-2023 Qiskit on IQM developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Fake backend for IQM's 6-qubit Deneb architecture.
+"""
+from iqm.iqm_client import QuantumArchitectureSpecification
+from iqm.qiskit_iqm.fake_backends.iqm_fake_backend import IQMErrorProfile, IQMFakeBackend
+
+
+def IQMFakeDeneb() -> IQMFakeBackend:
+    """Return IQMFakeBackend instance representing IQM's Deneb architecture.
+    Note: swaps are used in place of the move gates to enable the (noisy) simulation with AerSimulator.
+    """
+    MOVE = "swap"
```

Comment:
Yes, good idea. It is a bit awkward that we need to include the "swap" and "move" in the operations of IQMFakeDeneb (former for the noise model and latter as native gates of transpiler).

Unless someone figures better way, I'll keep it and add a more detailed comment.

Reply:
What is the reason to define the MOVE as swap and not as the MoveGate that has been implemented with a mathematical definition (being a swap matrix)? The MoveGate should implement any missing features to allow usage in the AerSimulator so that it runs the same code and the transpile pass Move to Swaps isn't needed.

Reply:
Initially, I tried to do as you suggest but didn't succeed. I tried chancing the _define() of MoveGate to swap instead of the three cnots that it is currently. Then I used only 'move' in the noise model, but got error that 'move' is not recognized by AerSimulator.

Reply:
I looked into this a bit further. The Aer simulator does not use the Gate definition to determine the unitary. I recommend to transpile MOVE gates to unitary gates where the unitary is a static field of the MoveGate class. And transpile to Move-to-Unitary following https://qiskit.github.io/qiskit-aer/tutorials/4_custom_gate_noise.html

Reply:
Yeah, I'm not sure if we just messaged past each other, but I think the better implementation is to have a static unitary field in MoveGate (nothing to do with the normal Qiskit API) and use that to transpile the MoveGates in the DAG to unitary gates that use the static field. That way, if we adjust the MoveGate description, the simulator can automatically pick it up and we don't need to remember to adjust it there. 

Given `noise_model = NoiseModel(basis_gates=["r", "cz", "unitary"])`
For example:
```
move_as_unitary = QuantumCircuit(2)
move_as_unitary.unitary(MoveGate.unitary,0, 1)
dag.substitute_node_with_dag(node, circuit_to_dag(move_as_unitary))
```

Alternatively you could do something like (not sure about compositionality of these functions)
```
unitary = Operator(dag_to_circuit(DAGCircuit)).to_matrix() 
```
Or
```
unitary = MoveGate(node.args).to_matrix()
```

That way, the simulator is not relying on the SWAP gate implementation specifically, but uses the specifics of the MoveGate class.

Reply:
Great, I was just going to ask exactly this. Do you prefer me adding static unitary to MoveGate or using one of the bottom cells to make the unitary in the IQMFakeBackend class?

Reply:
I pushed commit where I added unitary to MoveGate and use it for the noisy implementation.